### ISSUE TITLE
Syntax highlighting plugin for editor

### ISSUE THREAD

**Issue by manuwell** [Orange-OpenSource/hurl#983]
Hey guys, do you know if is there some plugin with syntax highlighting for hurl lang? 

I am a vim user, and didn't found anything. Even vscode doesn't have a plugin. 
What you use to syntax highlighting `.hurl` files?

**Comment by jcamiel** [comment]
Hi,

Unfortunately, Hurl tooling support needs more work now. There is some very basic syntax highlighting files for vim and Jetbrain's IDEs here: https://github.com/Orange-OpenSource/hurl/tree/master/contrib. 

We know we need to work on it (for VSCode for instance) external contributions are welcome of course!

**Comment by manuwell** [comment]
You are all great. I'll see if I can help with something. 

thanks a lot \o/

**Comment by didier-contribs** [comment]
I found minutes ago `https://github.com/fourjay/vim-hurl`, very nice syntax highlighter for vim

**Comment by jcamiel** [comment]
@didier-contribs Yes, you can try (and improve!) also this one https://github.com/Orange-OpenSource/hurl/tree/master/contrib/vim

**Comment by pfeiferj** [comment]
I just created a tree-sitter parser for the file format, it should be pretty accurate. I followed the grammar file pretty closely, but there could be some mistakes from not understanding the nuances of tree-sitter and/or bad find and replace.
https://github.com/pfeiferj/tree-sitter-hurl

**Comment by pfeiferj** [comment]
![Screenshot from 2022-12-21 01-54-33](https://user-images.githubusercontent.com/9106735/208840628-9e59db6c-47e6-492e-af23-fad608efb08a.png)
Here's a screenshot of the tree-sitter parser highlighting in my neovim instance

**Comment by pfeiferj** [comment]
![Screenshot from 2022-12-22 17-30-04](https://user-images.githubusercontent.com/9106735/209237001-94876b5a-147f-449b-9484-5f53fb4dcf1e.png)
Getting some promising results from using the tree-sitter parser in a vs-code extension as a semantic token lsp provider, may be able to publish it soon

**Comment by pfeiferj** [comment]
published the vs-code extension: https://marketplace.visualstudio.com/items?itemName=JacobPfeifer.pfeifer-hurl

**Comment by jcamiel** [comment]
Extra work @pfeiferj! Do you have any git repo for this if we want to contribute (when there is slight grammar modification)? Many thanks for this!

**Comment by pfeiferj** [comment]
The plugin uses the tree-sitter parser, so two separate repos
So the grammar is in: https://github.com/pfeiferj/tree-sitter-hurl
The plugin is: https://github.com/pfeiferj/vscode-hurl

**Comment by pfeiferj** [comment]
i've also got a neovim plugin going, although people can use the tree-sitter-parser in neovim without the plugin, just makes it a little easier. 
https://github.com/pfeiferj/nvim-hurl

**Comment by jellydn** [comment]
Hey everyone,

I’ve created two new tools to enhance your Hurl experience:

	1.	Neovim Plugin: [jellydn/hurl.nvim](https://github.com/jellydn/hurl.nvim)
	2.	VS Code Extension: [jellydn/vscode-hurl-runner](https://github.com/jellydn/vscode-hurl-runner)

Check them out and let me know what you think! Your feedback and suggestions are welcome.

Cheers,

**Comment by pfeiferj** [comment]
> Hey everyone,
> 
> I’ve created two new tools to enhance your Hurl experience:
> 
> 1. Neovim Plugin: [jellydn/hurl.nvim](https://github.com/jellydn/hurl.nvim)
> 2. VS Code Extension: [jellydn/vscode-hurl-runner](https://github.com/jellydn/vscode-hurl-runner)
> 
> Check them out and let me know what you think! Your feedback and suggestions are welcome.
> 
> Cheers,

Neat, I’ve more or less deprecated my neovim plugin, I’ve gotten my treesitter implementation into nvim-treesitter and I never really got a good runner going in my plugin, so I’ll definitely point people to your plugin over there. Kinda in the same place with my vscode plugin, I’ll probably deprecate the runner in my vscode plugin and point towards yours. I don’t currently use hurl for my job so my maintanence has been a bit slow on my plugins. I’ll have to test out the highlighting on your vscode plugin, not sure how well the regex highlighting performs vs the treesitter lsp highlighting. Ideally I wouldn’t mind deprecating my entire plugin in favor of one with a more active maintainer, and the highlighting would be the only thing that might prevent me from doing so

**Comment by jellydn** [comment]
Great. Thanks @pfeiferj 

> I’ll have to test out the highlighting on your vscode plugin, not sure how well the regex highlighting performs vs the treesitter lsp highlighting

Your treesitter plugin is better than my VS Code plugin however I believe the hurl syntax is lightweight. I tried to use web-tree-sitter but doesn't work with the latest treesitter version so I chose the current approach.

**Comment by pfeiferj** [comment]
> Your treesitter plugin is better than my VS Code plugin however I believe the hurl syntax is lightweight. I tried to use web-tree-sitter but doesn't work with the latest treesitter version so I chose the current approach. 

Gotcha, I'll just update mine to only focus on syntax highlighting and make it clear that additional functionality can be added with other extensions.

**Comment by Jiehong** [comment]
Referencing a plugin for Intellij: https://plugins.jetbrains.com/plugin/24735-hurl

**Comment by vincentbernat** [comment]
There is also [hurl-mode](https://github.com/JasZhe/hurl-mode), which provides more features than the one in contrib.

Out of curiosity, how is the syntax highlighting done in the documentation?

**Comment by jcamiel** [comment]
There is a "companion" tool for Hurl which is `hurlfmt`. It allows to:
- convert a `.hurl` file to JSON: if you want to export Hurl file to another format or if you need access to Hurl AST
- convert a `.hurl` file to HTML: that's how the syntax coloring is done in the docs
- lint `.hurl` file
- convert a curl command to a Hurl file
